Add `--v` flag to optionally include versions in `requirements.txt`

AnasImloul opened the pull request:
### PR Summary

This PR introduces a `--v` flag to the installation script, allowing users to optionally include the installed package versions in the requirements.txt file.

When the `--v` flag is used, the script will record the exact versions of the installed packages; otherwise, only the package names will be listed without versions.

AnasImloul commented on pwp/commands/__init__.py:
```diff
@@ -0,0 +1,2 @@
+from .install import pwp_install
```
I have added a folder `/commands` to reduce the load on `main.py`

AnasImloul commented on pwp/commands/install.py:
```diff
@@ -0,0 +1,60 @@
+        return None
+
+
+def create_or_update_requirements(packages: dict[str, str | None]):
```
`create_or_update_requirements` now takes packages as a `dict<name, version>` instead of a `list`, where `version` can be null.

merouanezouaid reviewed (changes requested):
install works perfectly fine..

for the versioned packages when uninstalling it can't detect if the package already exists in `requirements.txt` , and also sometimes when installing it it duplicates

Steps to reproduce:

```bash
pwp install numpy==2.0.1 # works fine and adds the package
pwp uninstall numpy==2.0.1 # doesn't remove the package from the file 
pwp install numpy==2.0.1 # package files gets added once again (same name 2 times)
```
output for second command:
```
Found existing installation: numpy 2.0.1
Uninstalling numpy-2.0.1:
  Successfully uninstalled numpy-2.0.1
Packages not found in requirements.txt: numpy==2.0.1````

AnasImloul commented:
Hi @merouanezouaid ,

I've fixed the bug you mentioned. Additionally, I identified another issue where the program did not alert if the user provided an illegal argument after the install or uninstall commands. For example, if an illegal flag like --g was added, the program previously treated it as a package rather than raising an error.

To address this, I've added a more robust argument parser to the program. This ensures that invalid flags or arguments are correctly identified and the user is notified with an appropriate error message.

merouanezouaid commented:
LGTM!
